src/ccextractor.c

```diff
@@ -457,3 +422,80 @@ int main(int argc, char *argv[])
 	}
 	return ret ? EXIT_OK : EXIT_NO_CAPTIONS;
 }
+
+struct ccx_s_options* api_init_options(){
+    init_options(&ccx_options);
+    return &ccx_options;
+}
+
+void checking_configuration_file(struct ccx_s_options api_options){
+    parse_configuration(&api_options);
+}   
+
+int compile_params(struct ccx_s_options *api_options,int argc){
+    
+    api_options->myarguments = realloc(api_options->myarguments, (api_options->argument_count+1) * sizeof *api_options->myarguments);
+      api_options->myarguments[api_options->argument_count] = malloc(strlen("./ccextractor")+1);
```

Comment:
What happens if you do not call the binary from the folder, but with a full path? e.g. calling it with `/home/foo/ccextractor/linux/ccextractor` instead of `./ccextractor`?